oterm: fix runtime error

gaelj opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
 Fix and update oterm by adding new python package `textualeffects`

- oterm: fix runtime error: missing textualeffects (#366021)
- ~~oterm: 0.6.9 -> 0.7.2~~ updated by the bot to 0.7.3 in the meantime
- textualeffects: init at 0.1.3

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

@suhr 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

suhr commented:
Please remove me from `meta.maintainers` in the `oterm` package.

gaelj commented:
fixed conflict after auto-update to 0.7.3 in the meantime

gaelj commented on pkgs/development/python-modules/textualeffects/default.nix:
```diff
@@ -0,0 +1,42 @@
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  python3Packages,
```
It's needed for `hatchling`

natsukium replied:
`hatchling` and `textualeffects` are both python modules, so we don't need python3Packages here.

gaelj replied:
I don't know how to fix this then...

![image](https://github.com/user-attachments/assets/2b5cbfb2-7f2e-441f-bcf6-952ad07fa1dd)

and

![image](https://github.com/user-attachments/assets/020e8a8a-cec5-4c96-8479-0a7240e6e780)

suhr replied:
Just replace `python3Packages` with `hatchling`.

gaelj commented:
Am I expected to close the conversations or should I let you do it ? From my point of view all requested changes were addressed.

gaelj commented:
@natsukium @suhr 

I just squashed the fixup commits.

Also, upstream has just released `0.8.0` and it requires some additional modifications to build (add of python lib `mcp` and update of `pydantic-settings`). I have already prepared these changes in a separate [branch](https://github.com/gaelj/nixpkgs/tree/oterm-0.8.0).

Should I wait until this PR is merged or should I merge `0.8.0` into this branch / PR ?

suhr commented:
> Should I wait until this PR is merged or should I merge 0.8.0 into this branch / PR ?

Creating a new one is probably a good idea.

gaelj commented:
@natsukium what can I do to get this merged ?

gaelj committed:
oterm: fix runtime error: missing textualeffects (#366021)

gaelj committed:
textualeffects: init at 0.1.3

textualeffects: integrate code-review comments

textualeffects: integrate code-review comments

gaelj committed:
maintainers: add gaelj

gaelj committed:
oterm: assign gaelj as maintainer
